src/Controls/src/BindingSourceGen/SetterBuilder.cs

```diff
@@ -28,32 +25,47 @@ public sealed record Setter(string[] PatternMatchingExpressions, string Assignme
 
     private sealed class SetterBuilder
     {
-        private readonly string _sourceVariableName;
+        private readonly bool _considerAllReferenceTypesPotentiallyNullable;
         private readonly string _assignedValueExpression;
 
         private string _expression;
         private int _variableCounter = 0;
         private List<string>? _patternMatching;
+
+        private IPathPart? _currentPart;
         private IPathPart? _previousPart;
+        private IPathPart _sourcePart;
 
-        public SetterBuilder(string sourceVariableName, string assignedValueExpression)
+        public SetterBuilder(bool considerAllReferenceTypesPotentiallyNullable, string sourceVariableName, TypeDescription sourceTypeDescription, string assignedValueExpression)
         {
-            _sourceVariableName = sourceVariableName;
+            _considerAllReferenceTypesPotentiallyNullable = considerAllReferenceTypesPotentiallyNullable;
             _assignedValueExpression = assignedValueExpression;
 
+            _sourcePart = new MemberAccess(sourceVariableName, sourceTypeDescription.IsValueType, sourceTypeDescription.IsValueType && sourceTypeDescription.IsNullable);
+            _currentPart = _sourcePart;
+
             _expression = sourceVariableName;
         }
 
         public void AddPart(IPathPart nextPart)
         {
-            _previousPart = HandlePreviousPart(nextPart);
+            var newPart = HandleCurrentPart(nextPart);
+            _previousPart = _currentPart;
+            _currentPart = newPart;
+
         }
```

Comment:
Thinking out loud: this needs refactoring, the `nextPart`, `newPart`, `_currentPart`, `_previousPart` naming is confusing.